### ISSUE TITLE
Remove all the leftover of OmpAttribute

### ISSUE THREAD

**Issue by ouankou** [passlab/rexompiler#130]
`OmpAttribute` is the original IR of the ROSE OpenMP parser. We don't need that anymore since we use ompparser, whose IR is `OpenMPIR`. Besides the OpenMP AST constructor, we also need to clean up all the code of OmpAttribute in other locations.

**Comment by ouankou** [comment]
According to the failure of a [recent CI test](https://github.com/passlab/rexompiler/actions/runs/3584525832/jobs/6031314967), the EDG submodule also depends on `OmpAgttribute.h`. We need to investigate whether this dependency can be removed and how to remove it from the EDG source code.

**Comment by yanyh15** [comment]
I have renamed the OmpAttribute.h/C file to ompSupport.h/C, and removed OmpAttribute class and its member functions, but leaving the enum definition there. I also removed OmpSupport.h file in ompLowering folder since it is just a header that include other headers. I am fixing this and will push.

**Comment by yanyh15** [comment]
@ouankou I have a fix in EDG repo and now it is pushed to the master branch. Which branch is used for REX? Could you update the rex gitmodule to make sure that REX pull the needed EDG?

**Comment by yanyh15** [comment]
@ouankou I have pushed latest fix for REX and CI is running, let us see whether it will pass.

**Comment by ouankou** [comment]
> @ouankou I have pushed latest fix for REX and CI is running, let us see whether it will pass.

Thanks, Dr. Yan. The submodule is tied to a specific commit, regardless of which branch the commit is on.

**Comment by yanyh15** [comment]
@ouankou what is the status of this issue? are you fixing it along with reverting upir?

**Comment by ouankou** [comment]
Yes, I'm also cleaning up the code while reverting UPIR. All the code used OmpAttribute has been removed by your previous commits except one: https://github.com/passlab/rexompiler/blob/a6bd2df2441e2a1d8b62df20b7258fe3cc38b6ba/src/backend/unparser/FortranCodeGeneration/unparseFortran_statements.C#L4050

There are several other leftover comments related to `OmpAttribute`, I've removed them.

I fixed this by determining whether the leading keyword is `omp`. However, it triggered a bug that unparsing `omp parallel` produces two `omp end parallel`. I'm still working on this. and it should be done this week.

**Comment by ouankou** [comment]
This unparsing bug was caused by some other unknown problem instead of `OmpAttributeList` here.

Given input:
```fortran
      program main
      include 'omp_lib.h'
      integer i
      integer item(10)
      external process
      do i = 1, 10
        item(i) = 11-i
      enddo  

!$omp parallel
!$omp single
      print *, 'using ', omp_get_num_threads(), ' threads' 
      do i = 1, 10
!$omp task 
      call process(i,item)
!$omp end task
      enddo  
!$omp end single
!$omp end parallel
      end 

******************************************
      subroutine process(input,item)
      include 'omp_lib.h'
      integer input
      integer item(10)
      print *, 'idx ', input,'-> ', item(input)
      print *, 'by thread ', omp_get_thread_num()
      end
```

`!pragma omp end parallel` or `!$omp omp end parallel` will be inserted right above `!$omp end single`. Neither should happen.

The output from code above:
```fortran
      PROGRAM main
      include "omp_lib.h"
      INTEGER :: i
      INTEGER, DIMENSION(10) :: item
      external :: process
      DO i = 1, 10
      item(i) = 11 - i
      END DO
      
!$omp parallel 
      
!$omp single 
      PRINT *, 'using ',omp_get_num_threads(),' threads'
      DO i = 1, 10
      
!$omp task 
      CALL process(i,item)
!$omp end task 
      END DO
      !pragma omp end parallel 
!$omp end single 
******************************************
!$omp end parallel
      END PROGRAM 

      SUBROUTINE process(input,item)
      include "omp_lib.h"
      INTEGER :: input
      INTEGER, DIMENSION(10) :: item
      PRINT *, 'idx ',input,'-> ',item(input)
      PRINT *, 'by thread ',omp_get_thread_num()
      END SUBROUTINE
```

**Comment by ouankou** [comment]
The incorrect codegen above was tracked and fixed in #134. All code related to OmpAttribute has been removed except some text in the document [`docs/Rose/OpenMPsupport.tex`](https://github.com/passlab/rexompiler/blob/dev/docs/Rose/OpenMPsupport.tex).
